src/dependency/Pyegi/Pyegi/installer.py

```diff
@@ -0,0 +1,169 @@
+import os
+from os.path import exists
+import sys
+import toml
+from datetime import datetime as dt1
+import shutil
+import csv
+import json
+
+start_time = dt1.now()
+print("\nStart time: " + str(start_time) + "\n")
+
+temp_dir = os.path.dirname(__file__) + "/temp/"
+commons_dir = os.path.dirname(__file__) + "/commons/"
+dependency_dir = os.path.dirname(os.path.dirname(__file__)) + "/"
+scriptsPath = dependency_dir + "PythonScripts/"
+# system_inputs = sys.argv
+
+scripts_names = [
+    name
+    for name in os.listdir(scriptsPath)
+    if os.path.isdir(os.path.join(scriptsPath, name))
+]
+
+
+def create_dirs(path):
+    parent, _ = os.path.split(path)
+    if not exists(parent):
+        os.makedirs(parent)
+
+
+def install_pkg(script):
+    print(f"Processing {script} dependencies...")
+    script_path = scriptsPath + script + "/"
+    if exists(script_path + "pyproject.toml"):
+        os.chdir(script_path)
+        os.system("poetry lock")
+        src = script_path + "pyproject.toml"
+        dst = temp_dir + "pyproject.toml"
+        shutil.copyfile(src, dst)
+        os.chdir(temp_dir)
+        os.system("poetry lock")
```

Comment:
which one?

Reply:
2nd

Reply:
I'm not following. There isn't a second copy command.

Reply:
I was mentioning the lock file. we called poetry lock once. we can just copy it alongside pyproject.
![image](https://user-images.githubusercontent.com/25867589/181258857-e3135b1f-1692-4ba3-91ca-d9aeb61e30f0.png)

Reply:
I talked about this before but there are some files in each venv which contain the path of their parent directory. So, I advise against copying .venv directory to the temp directory.

Reply:
I didn't mean the `.venv` folder. Just the lock and pyproject file. There isn't any disturbing relative paths in the lock file itself.
But now I can why you ran it twice. You wanted to create the `.venv` folder. So, we should keep the second `poetry lock`. I think if you copy the lock file to the temp dir and then you call `poetry lock --no-update` instead of `poetry lock`, the process would go faster.

Reply:
yeah, this is worth trying